src/System.Private.ServiceModel/src/System/ServiceModel/Channels/RequestChannel.cs

```diff
@@ -196,10 +197,15 @@ private void ReleaseRequest(IRequestBase request)
                 _outstandingRequests.Remove(request);
                 if (_outstandingRequests.Count == 0)
                 {
-                    if (!_closed && _closedTcs != null)
+                    // When we are closed or closing, _closedTcs is managed by the close logic.
+                    if (!_closed && State != CommunicationState.Closing)
                     {
-                        _closedTcs.TrySetResult(null);
-                        _closedTcs = null;
+                        // Protect against close altering _closedTcs concurrently
+                        var closedTcs = Interlocked.CompareExchange(ref _closedTcs, null, _closedTcs);
+                        if (closedTcs != null)
+                        {
+                            closedTcs.TrySetResult(null);
```

Comment:
My read is that if the close succeeded in its own TrySetResult (and TCS was not yet null -- which should not be possible given the state checks above), the TrySetResult here would be a NOP.  And nothing checks the return code for TrySetResult.  Translation: I think we're good.